website/docs/spec/index.md

```diff
@@ -111,12 +111,14 @@ All MCAP records are serialized as follows:
 
 Record type is a single byte opcode, and record content length is a uint64 value.
 
-Records may be extended by adding new fields at the end of existing fields. Readers should ignore any unknown fields.
+Future changes to this specification may extend records by adding new fields at the end of existing fields. Readers should ignore any unknown fields.
 
 > The Footer and Message records will not be extended, since their formats do not allow for backward-compatible size changes.
 
 Each record definition below contains a `Type` column. See the [Serialization](#serialization) section on how to serialize each type.
 
+Record content may end before all known fields have been serialized. Readers should treat a missing field as having that field type's [zero value](#zero-values).
```

Comment:
if we are going to make this statement, we should probably be explicit that records must still be fully formed, even if they are missing fields (i.e. records must match their stated length). otherwise the file should be considered corrupt.